workitem/expression_compiler.go

```diff
@@ -273,6 +273,15 @@ func (c *expressionCompiler) expressionRefersToJoinedData(e criteria.Expression)
 	return nil, false
 }
 
+// ensureJoinTable make sure the table is joined in the query
+func (c *expressionCompiler) ensureJoinTable(tableName string) {
```

Comment:
We don't need this function and it most certainly doesn't have to contain a loop. The `c.joins` member is a `map` so you can just do `c.joins[tableName].Active = true`. There's no need to loop over all joins.